src/primitives/torus.jl

```diff
@@ -0,0 +1,54 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Torus(majorRadius, minorRadius)
+
+A torus with `majorRadius` and `minorRadius` with the z axis as its axis of 
+rotation and the xy plane as its plane of reflection. 
```

Comment:
Worth adding a link to the Wikipedia definition like we do with other geometries.